llama-index-integrations/llms/llama-index-llms-vllm/llama_index/llms/vllm/base.py

```diff
@@ -247,12 +247,12 @@ def _model_kwargs(self) -> Dict[str, Any]:
         }
         return {**base_kwargs}
 
-    def __del__(self) -> None:
+    @atexit.register
+    def close():    
         import torch
         import gc
-
         if torch.cuda.is_available():
-            del self._client
+            _client = None
```

Comment:
shouldn't this be `self._client` ? `_client` isn't a variable in scope right, so this is just creating a new variable?

Reply:
self._client does not seem to be in scope (when using @atexit) so I used _client instead. You are correct that this is simply an unused local variable at this point. So, the references to _client can be deleted without any negative consequences.

Reply:
Logan - Can you help move this PR along ?  The lint issue was resolved (by removing some blanks).